### PR TITLE
89 a specific verses explanation including roots linguistics and reflections

### PR DESCRIPTION
**Pull request by ibrahim681**

### CODE REVIEW

**Comment by osamaramihafez** [comment]
To fix conflicts, merge locally from master and accept the changes that are coming from master. You should not change the schema but can add your import inside of index.js

File: routes/verseInfo.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ *        type: integer
+ *        required: true
+ *        example: 1
+ *  responses:
+ *    200:
+ *      description: The verse's complete information.
+ *      schema:
+ *        type: array
+ *        items:
+ *          $ref: '#/definitions/Verse'
```
Comment (osamaramihafez) [inline routes/verseInfo.js:27]: You should create a definition specifically for this item, see #/definitions/VerseWordMeaning for an example.

File: model/verseInfo.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
+const { parseHttpResponse } = require("selenium-webdriver/http");
+const utils = require("./utils");
+
```
Comment (osamaramihafez) [inline model/verseInfo.js:3]: We are returning a new type of object (which contains arrays of other objects). Please add a definition for this schema object and call it something like VerseInformation

File: tests/setup.js [hunk 3]
```diff
@@ -62,6 +68,61 @@ const seedData = {
...
+    VerseWord: [
+        {
+            verse_word_id: 1,
+            verse_id: 1,
+            word_id: 1,
+            visible: false,
+            word_explaination: "In the Name of Allah",
+        },
+    ],
+    Mufasir: [
```
Comment (osamaramihafez) [inline tests/setup.js:104]: Mufasir exists twice

File: tests/setup.js [hunk 4]
```diff
@@ -62,6 +68,61 @@ const seedData = {
             root_id: 1,
             root_word: "س م و",
         },
+        {
+            root_id: 2,
+            root_word: "ز ح ز ح"
+        }
+    ],
+    RootMeaning: [
```
Comment (osamaramihafez) [inline tests/setup.js:76]: Exists twice, remove this one.

File: tests/setup.js (hunk 3 above)
Comment (osamaramihafez) [reply tests/setup.js:104]: Also make sure no other tables have been duplicated.

File: tests/verse-info-tests.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+}
+function checkRootMatch(t1, vw, aw) {
+    expect(t1.root_id).toEqual(aw.root_id)
+    expect(t1.verse_id).toEqual(vw.verse_id)
+    expect(t1.word_id).toEqual(vw.word_id)
+}
+
+module.exports = {
+    verseInfoTests: verseInfoTests,
+};
```
Comment (osamaramihafez) [inline tests/verse-info-tests.js:44]: Kind of petty, but try to add a newline character at the end of every document. I forget why it's important, but github keeps highlighting it anyways.

File: tests/verse-info-tests.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    expect(t1.tafsir_id).toEqual(t2.tafsir_id);
+    expect(t1.content).toEqual(t2.content);
+    expect(t1.verse_id).toEqual(t2.verse_id);
+}
+function checkRootMatch(t1, vw, aw) {
+    expect(t1.root_id).toEqual(aw.root_id)
+    expect(t1.verse_id).toEqual(vw.verse_id)
+    expect(t1.word_id).toEqual(vw.word_id)
+}
+
```
Comment (osamaramihafez) [inline tests/verse-info-tests.js:41]: Should we also check for an Arabic word match?

**Review by osamaramihafez — CHANGES_REQUESTED** [review]

File: db/schema.sql [hunk 7]
```diff
@@ -1,120 +1,137 @@
...
-    lesson_date DATE NOT NULL,
-    source TEXT NOT NULL,
-    surah_id INTEGER,
-
-    FOREIGN KEY (surah_id) REFERENCES Surah(surah_id) ON DELETE CASCADE ON UPDATE CASCADE
-);
+DROP TABLE IF EXISTS Surah CASCADE;
+
+CREATE TABLE
+    IF NOT EXISTS Surah (
```
[the remark this answers is not in the document]
Comment (ibrahim681) [reply db/schema.sql:4]: not sure how to change it, I tried code>preferences>settings Prettier, but didn't seem to work🥲

[the remark this answers is not in the document]
Comment (osamaramihafez) [reply db/schema.sql:4]: Don't worry, it's fine for now

File: model/verseInfo.js [hunk 8]
```diff
@@ -0,0 +1,146 @@
+const { parseHttpResponse } = require("selenium-webdriver/http");
```
Comment (osamaramihafez) [inline model/verseInfo.js:1]: Don't import selenium... why? lol

File: routes/verseInfo.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
+const router = require("express").Router();
+const verseInfo = require("../model/verseInfo");
+const utils = require("./utils");
+const Ucreate = require("../model/utils");
```
Comment (osamaramihafez) [inline routes/verseInfo.js:4]: Ucreate?

File: tests/setup.js [hunk 10]
```diff
@@ -49,43 +49,88 @@ const seedData = {
...
     ArabicWord: [
         {
             word_id: 1,
             word: "بِسْمِ",
-            root_id: 1,
+            root_id: 1
         },
+        {
+            word_id: 2,
+            word: "بِسْمِ",
```
Comment (osamaramihafez) [inline tests/setup.js:84]: Let's put in another arabic word, perhaps "الرَّحِيمِ"

File: tests/setup.js [hunk 11]
```diff
@@ -49,43 +49,88 @@ const seedData = {
...
+        }
+    ],
+    RootMeaning: [
+        {
+            meaning_id: 1,
+            root_id: 1,
+            meaning: "name",
         },
     ],
+    ArabicWord: [
```
Comment (osamaramihafez) [inline tests/setup.js:127]: Arabic word is still duplicated

File: tests/verse-info-tests.js [hunk 12]
```diff
@@ -0,0 +1,46 @@
+const faker = require("faker");
```
Comment (osamaramihafez) [inline tests/verse-info-tests.js:1]: Remove faker if not being used

**Review by osamaramihafez — DISMISSED** [review]
Also see other comments I've made

**Comment by osamaramihafez** [comment]
LGTM, ready to merge! Good work 👍🏼

**Pull request merged**